airflow/providers/google/cloud/hooks/dataflow.py

```diff
@@ -278,18 +296,20 @@ def _check_dataflow_job_state(self, job) -> bool:
         :rtype: bool
         :raise: Exception
         """
-        if DataflowJobStatus.JOB_STATE_DONE == job["currentState"]:
+        if self._wait_until_finished is None:
+            wait_until_finished = job['type'] != DataflowJobType.JOB_TYPE_STREAMING
+        else:
+            wait_until_finished = self._wait_until_finished
+
+        if job['currentState'] == DataflowJobStatus.JOB_STATE_DONE:
```

Comment:
DRAINED, UPDATED also needs to be in this list. They are alos terminal states.

(https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.jobs#jobstate)

Reply:
Thanks! Just added it here:  https://github.com/apache/airflow/pull/11726/commits/d590f9221967b73d43780d9aeb35cd96b2d2ca28